### PR TITLE
Support change rocksdb options from console

### LINKED ISSUE #835: Support change rocksdb options dynamicly
Extension of #724. 

My idea is that we can change all our rocksdb options in console by GflagsManger and when stoarge and meta found them updated, apply them to rocksdb.

For those complicate options, we can organize them in a json-like string, so we can update only one field without bothering others.

### PR DESCRIPTION
**Pull request by critical27**
1. According to @whitewum's suggestion, I change keyword from VARIABLES to CONFIGS.
2. Now we support change rocksdb options from console, Meta Client will retrieve newest options periodically, and apply them to rocksdb if it is updated.

For those options we can change dynamically, we can update them in console like this:
`update configs storage:rocksdb_column_family_options = { disable_auto_compactions = false , level0_file_num_compaction_trigger = 10 }`

close #835

### CODE REVIEW

**Comment by dangleptr** [comment]
Very practical feature!

**Comment by critical27** [comment]
Jenkins go

**Comment by critical27** [comment]
Jenkins go

**Comment by critical27** [comment]
Jenkins go

**Comment by critical27** [comment]
Jenkins go

**Comment by whitewum** [comment]
please info @Amber1990Zhang to update the doc

**Comment by critical27** [comment]
> please info @Amber1990Zhang to update the doc

Sure, I've told her.

File: src/kvstore/PartManager.cpp [hunk 1]
```diff
@@ -83,6 +83,58 @@ void MetaServerBasedPartManager::onSpaceRemoved(GraphSpaceID spaceId) {
...
+    for (const auto& option : options) {
+        if (supportedOpt.find(option.first) != supportedOpt.end()) {
+            opt[option.first] = option.second;
+        } else if (supportedDbOpt.find(option.first) != supportedDbOpt.end()) {
+            dbOpt[option.first] = option.second;
+        }
+    }
+
+    if (handler_ != nullptr) {
+        if (opt.size() > 0) {
```
Comment (yixinglu) [inline src/kvstore/PartManager.cpp]: Replace `opt.size() > 0` with `!opt.empty()`

File: src/kvstore/PartManager.cpp [hunk 2]
```diff
@@ -83,6 +83,58 @@ void MetaServerBasedPartManager::onSpaceRemoved(GraphSpaceID spaceId) {
...
+            handler_->updateSpaceOption(spaceId, opt, false);
+        }
+        if (dbOpt.size() > 0) {
```
Comment (yixinglu) [inline src/kvstore/PartManager.cpp]: ditto

File: src/meta/test/ConfigManTest.cpp [hunk 3]
```diff
@@ -171,6 +183,50 @@ TEST(ConfigManTest, ConfigProcessorTest) {
...
         ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, resp.get_code());
     }
+    // update the nested config
+    {
+        cpp2::ConfigItem updated;
+        updated.set_module(cpp2::ConfigModule::META);
+        updated.set_name("nested");
+        updated.set_type(cpp2::ConfigType::NESTED);
+        updated.set_mode(cpp2::ConfigMode::MUTABLE);
+        // upate from consle as format of update list
```
Comment (yixinglu) [inline src/meta/test/ConfigManTest.cpp]: typo: `update` and `console`

File: src/common/base/Configuration.h [hunk 4]
```diff
@@ -32,6 +32,11 @@ class Configuration final {
      * Parse from a string buffer
      */
     Status MUST_USE_RESULT parseFromString(const std::string &content);
+
+    std::string dumpToString();
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/common/base/Configuration.h]: Good suggestion.

File: src/common/base/Configuration.h [hunk 5]
```diff
@@ -32,6 +32,11 @@ class Configuration final {
...
+
+    std::string dumpToPrettyString();
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/common/base/Configuration.h]: Good suggestion.

File: src/common/base/Configuration.cpp [hunk 6]
```diff
@@ -144,6 +160,17 @@ Status Configuration::fetchAsSubConf(const char *key, Configuration &subconf) co
 }
 
 
+Status Configuration::updateStringField(const char* key, const std::string& val) {
+    DCHECK(content_ != nullptr);
+    auto iter = content_->find(key);
+    if (iter == content_->items().end() || iter->second.isString()) {
```
[the remark this answers is not in the document]
Comment (critical27) [reply src/common/base/Configuration.cpp]: We will update if one of following is true: 
1. key is not in content_
2. key is in content_ and value is string.

File: src/meta/test/ConfigManTest.cpp (hunk 3 above)
Comment (critical27) [reply src/meta/test/ConfigManTest.cpp]: Good catch

**Pull request merged**